node/src/components/consensus/era_supervisor/era.rs

```diff
@@ -221,3 +285,130 @@ where
+        };
+        let report = {
+            let total = ALICE_STAKE + BOB_STAKE + CAROL_STAKE;
+            EraReport {
+                rewards: vec![
+                    (
+                        ALICE_PUBLIC_KEY.clone(),
+                        ALICE_STAKE * number_of_rounds_in_this_era * BLOCK_REWARD / total,
+                    ),
+                    (
+                        BOB_PUBLIC_KEY.clone(),
+                        BOB_STAKE * number_of_rounds_in_this_era * BLOCK_REWARD / total,
+                    ),
+                    (
+                        CAROL_PUBLIC_KEY.clone(),
+                        CAROL_STAKE * number_of_rounds_in_this_era * BLOCK_REWARD / total,
+                    ),
+                ]
+                .into_iter()
+                .collect(),
+                equivocators: vec![],
+                inactive_validators: vec![],
+            }
+        };
+        assert_eq!(report, era.create_report(tbd, time, &protocol_config));
+
+        // If Carol is inactive in this era, she doesn't get a reward.
+        let tbd = TerminalBlockData {
+            inactive_validators: vec![CAROL_PUBLIC_KEY.clone()],
+        };
+        let report = {
+            let total = ALICE_STAKE + BOB_STAKE;
+            EraReport {
+                rewards: vec![
+                    (
+                        ALICE_PUBLIC_KEY.clone(),
+                        ALICE_STAKE * number_of_rounds_in_this_era * BLOCK_REWARD / total,
+                    ),
+                    (
+                        BOB_PUBLIC_KEY.clone(),
+                        BOB_STAKE * number_of_rounds_in_this_era * BLOCK_REWARD / total,
+                    ),
+                ]
+                .into_iter()
+                .collect(),
+                equivocators: vec![],
+                inactive_validators: vec![CAROL_PUBLIC_KEY.clone()],
+            }
+        };
+        assert_eq!(report, era.create_report(tbd, time, &protocol_config));
+
+        // And if she is faulty, she doesn't get one either.
+        let tbd = TerminalBlockData {
+            inactive_validators: vec![],
+        };
+        era.add_accusations(&[CAROL_PUBLIC_KEY.clone()]);
+        let report = {
+            let total = ALICE_STAKE + BOB_STAKE;
+            EraReport {
+                rewards: vec![
+                    (
+                        ALICE_PUBLIC_KEY.clone(),
+                        ALICE_STAKE * number_of_rounds_in_this_era * BLOCK_REWARD / total,
+                    ),
+                    (
+                        BOB_PUBLIC_KEY.clone(),
+                        BOB_STAKE * number_of_rounds_in_this_era * BLOCK_REWARD / total,
+                    ),
+                ]
+                .into_iter()
+                .collect(),
+                equivocators: vec![CAROL_PUBLIC_KEY.clone()],
+                inactive_validators: vec![],
+            }
+        };
+        assert_eq!(report, era.create_report(tbd, time, &protocol_config));
+    }
```

Comment:
It looks like we're missing just one test case - where `total_rewards == 0`.

Reply:
Done in https://github.com/casper-network/casper-node/pull/1950/commits/8a8090c72da2779d476924de970514641e6f93e7.